cli/src/bin/phylum.rs

```diff
@@ -46,45 +47,56 @@ pub fn exit_error(error: Box<dyn std::error::Error>, message: impl AsRef<str>) -
     ExitCode::Generic.exit()
 }
 
+/// Construct an instance of `PhylumApi` given configuration, optional timeout, and whether we
+/// need API to ignore certificates.
+async fn api_factory(
+    config: &mut Config,
+    config_path: &Path,
+    timeout: Option<u64>,
+    ignore_certs: bool,
+) -> Result<PhylumApi> {
+    let api = PhylumApi::new(
+        &mut config.auth_info,
+        &config.connection.uri,
+        timeout,
+        ignore_certs,
+    )
+    .await
+    .context("Error creating client")?;
+
+    // PhylumApi may have had to log in, updating the auth info so we should save the config
+    save_config(config_path, &config).with_context(|| {
+        format!(
+            "Failed to save configuration to '{}'",
+            config_path.to_string_lossy()
+        )
+    })?;
+
+    Ok(api)
+}
+
 async fn handle_commands() -> CommandResult {
+    //
+    // Initialize clap app and read configuration.
+    //
+
     let app = phylum_cli::app::app()
         .arg_required_else_help(true)
         .subcommand_required(true);
     let app_name = app.get_name().to_string();
-    let ver = app.get_version().unwrap();
-
     // Required for printing help messages since `get_matches()` consumes `App`
     let app_helper = &mut app.clone();
-
+    let ver = app.get_version().unwrap();
```

Comment:
Why was this moved?

Reply:
Just to keep initialization-related/`app` unpacking things closer together.